socorro/cron/jobs/ftpscraper.py

```diff
@@ -271,14 +271,23 @@ def __init__(self, *args, **kwargs):
         )
 
     def download(self, url):
+        try:
+            return self._download_cache[url]
```

Comment:
The speed difference between try:except and `in` is negligible when doing only a few thousand of these. I did it like this to keep the cache local and not require changes outside that method. 

Instead, a cleaner solution altogether is to use a memoizer. I'll just switch to that instead.